db/functions/base.py

```diff
@@ -13,15 +13,46 @@
 """
 
 from abc import ABC, abstractmethod
+import warnings
 
-from sqlalchemy import column, not_, and_, or_, func, literal, cast, INTEGER
-from sqlalchemy.dialects.postgresql import array_agg
+from sqlalchemy import column, not_, and_, or_, func, literal
+from sqlalchemy.dialects.postgresql import array_agg, INTEGER, TEXT
+from sqlalchemy.sql import quoted_name
+from sqlalchemy.sql.functions import GenericFunction, concat
 
 from db.functions import hints
 from db.functions.exceptions import BadDBFunctionFormat
 
 
-def sa_call_sql_function(function_name, *parameters):
+def sa_call_sql_function(function_name, *parameters, return_type=None):
+    """
+    This function registers an SQL function with SQLAlchemy and generates
+    an expression to call it.
+
+    function_name: string giving a namespaced SQL function
+    *parameters:   these will be passed directly to the generated function.
+    return_type:   an SQLAlchemy type class
+    """
+    if return_type is None:
+        warnings.warn(
+            "sa_call_sql_function should be called with the return_type kwarg set"
+        )
+        # We can't use PostgresType since we don't want an engine here
+        return_type = TEXT
```

Comment:
> We can't use PostgresType since we don't want an engine here

I wonder what you mean by that. The types under `db.types` should completely replace all SQLAlchemy type usage in the rest of codebase. Sounds like you're running into some tech debt. I'd be willing to help solve this. Hoping this doesn't prove necessary, since feels like a serious regression.

Reply:
We can't get the SQLAlchemy class associated with a type without an engine. To get an engine inside this function seemed pretty contrived. I agree that's a bit irritating, but I didn't want to do a major refactor to get the engine in here (it would have to be passed from the calling model to be proper in our current design, but how should that happen for a `DBFunction` call?). So, I went with the hack. 

I think it's okay, since I consider all code that uses SQLAlchemy types to be temporary at this point.

Reply:
Okay, I see your later comment. Let me try to get that working.

Reply:
Okay, I've looked into this. Long story short, I don't want to do the amount of refactoring that would be necessary to fix this preexisting problem in this PR. I think we should instead 
1. make an issue to note the problem, and
2. consider whether it's worth refactoring at all, since we are likely going to get rid of SQLAlchemy anyway, and this will mean reworking how we handle types.